### ISSUE TITLE
fix stderr overloading

### ISSUE THREAD

**Issue by AndreasMadsen** [AndreasMadsen/immortal#9]
The `stderr` from the `pump` is saved in the `process` until the `pump` restart. But if a lot of `stderr` is outputted from the `pump` the `daemon` will die. Because memory limits are exited.
This is properly not a big problem since the `pump` will restart the `daemon` and then forcekill itself. But if the `pump` when intro a infinit loop like `while(true) process.stderr.write('a');` the event loop will be blocked and the `daemon` will nerver restart.

**Comment by AndreasMadsen** [comment]
This is actually more serious than I originally thought. Even in an async loop:

``` JavaScript
(function loop() {
  process.stderr.write('a');
  process.nextTick(loop);
})();
```

Because when the daemon dies the pump outputs to nowhere the OS will send it a SIGPIPE and the pump will die. Now both daemon and pump are dead (no restart process), and the child process will also summon with no parent (or until it writes something).

**So please don't make an error loop in the monitor, until this is solved ;)**

**Comment by AndreasMadsen** [comment]
The "simultaneous error in daemon and monitor" bug is fixed by 886b112187e8e5ad93d53be47c2b938dd55b5391, however it is a general fix for all simultaneous errors there might happen. This specific bug, could be handled in an even better way.

**Comment by AndreasMadsen** [comment]
Finally fixed by: 0543660680cb52f9a86a4f07e96bb0ac6ccb1079
